stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt

```diff
@@ -121,7 +129,24 @@ internal class ChatSocketServiceImpl private constructor(
         this.endpoint = endpoint
         this.apiKey = apiKey
         this.user = user
-        setupSocket()
+
```

Comment:
A small problem with this scenario:
1) Connection lost
2) Collapse app
3) Bring the app back to foreground
The state will be State.Disconnected instead of State.NetworkDisconnected, so the monitor will start running.

Reply:
Nice catch!